src/composables/useReportDownload.ts

```diff
@@ -1,42 +1,74 @@
 import jsonexport from 'jsonexport/dist';
 import pkg from '@/../package.json';
-import { getProposalVotes } from '@/helpers/snapshot';
+import { getProposalVotes, getProposal } from '@/helpers/snapshot';
 import { Vote } from '@/helpers/interfaces';
 
 export function useReportDownload() {
+  const isDownloadingVotes = ref(false);
+  const downloadProgress = ref(0);
+
   async function getCsvFile(
     data: any[] | Record<string, any>,
+    headers: string[],
     fileName: string
   ) {
-    const csv = await jsonexport(data);
+    const csv = await jsonexport(data, { headers });
     const link = document.createElement('a');
     link.setAttribute('href', `data:text/csv;charset=utf-8,${csv}`);
     link.setAttribute('download', `${fileName}.csv`);
     document.body.appendChild(link);
     link.click();
   }
 
-  async function getAllVotes(proposalId: string, space: string) {
+  async function getAllVotes(
+    proposalId: string,
+    space: string,
+    totalVotesCount: number
+  ) {
     let votes: Vote[] = [];
     let page = 0;
+    let createdPivot = 0;
     const pageSize = 1000;
-    while (votes.length === pageSize * page) {
-      const newVotes = await getProposalVotes(proposalId, {
+    let resultsSize = 0;
+    const maxPage = 5;
+    do {
+      let newVotes = await getProposalVotes(proposalId, {
```

Comment:
Should we use a try catch here?

Reply:
There's already a try/catch on the `getProposalVotes` side.

We could remove it, and move it to the UI side, to handle errors

Reply:
That will make the error bubble 2 layers upward, snapshot.ts > userReportDownload.ts > SpaceProposalVotesList.vue, and also require all components using `getProposalVotes` to implement their own try/catch

Reply:
Yes right, let's leave it as is I'd say